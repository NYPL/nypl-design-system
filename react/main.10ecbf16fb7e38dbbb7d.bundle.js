(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1292:function(n,o,c){"use strict";c.r(o);c(1293),c(1323),c(1335),c(1442),c(1485)},1448:function(n,o){},1449:function(n,o){},1450:function(n,o){},1451:function(n,o){},501:function(n,o,c){c(502),c(645),n.exports=c(1292)},564:function(n,o){}},[[501,1,2]]]);