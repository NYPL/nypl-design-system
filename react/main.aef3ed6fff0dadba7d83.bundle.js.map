{"version":3,"file":"main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}