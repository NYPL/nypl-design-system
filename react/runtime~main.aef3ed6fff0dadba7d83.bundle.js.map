{"version":3,"file":"runtime~main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///runtime~main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":"AAAA","sourceRoot":""}