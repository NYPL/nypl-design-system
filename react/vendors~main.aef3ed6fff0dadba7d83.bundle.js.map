{"version":3,"file":"vendors~main.aef3ed6fff0dadba7d83.bundle.js","sources":["webpack:///vendors~main.aef3ed6fff0dadba7d83.bundle.js"],"mappings":";AAAA","sourceRoot":""}