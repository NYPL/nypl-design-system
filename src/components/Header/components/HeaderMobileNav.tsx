import {
  Box,
  chakra,
  Flex,
  Spacer,
  useMultiStyleConfig,
} from "@chakra-ui/react";
import React from "react";

import Icon from "../../Icons/Icon";
import Link from "../../Link/Link";
import List from "../../List/List";
import Logo from "../../Logo/Logo";
import SimpleGrid from "../../Grid/SimpleGrid";
import gaUtils from "../utils/googleAnalyticsUtils";
import { siteNavLinks, upperNavLinks } from "../utils/headerUtils";

/**
 * This component renders the navigational list of links used to navigate
 * NYPL.org for mobile devices.
 */
const HeaderMobileNav = chakra(() => {
  const styles = useMultiStyleConfig("HeaderMobileNav", {});
  const listItems = siteNavLinks.map(({ href, text }) => (
    <Link
      href={href}
      key={text}
      onClick={() => gaUtils.trackEvent("Go to...", text)}
    >
      {text}
    </Link>
  ));

  return (
    <Box __css={styles}>
      <Flex>
        <Box>
          <Logo
            aria-label="NYPL Header Logo"
            decorative={false}
            name="nyplTextWhite"
            size="xsmall"
            title="NYPL Header Logo"
            __css={styles.logo}
          />
        </Box>
        <Spacer />
        <nav aria-label="Header mobile links">
          <List
            id="header-mobile-nav"
            listItems={listItems}
            noStyling
            type="ul"
            __css={styles.sideNav}
          />
        </nav>
      </Flex>
      <SimpleGrid gap="0" data-testid="bottomLinks" __css={styles.bottomLinks}>
        <Link
          href={upperNavLinks.libraryCard.href}
          borderTop="1px solid rgb(54, 54, 54)"
          borderRight="1px solid rgb(54, 54, 54)"
          gridColumn="1 / span 1"
          onClick={() =>
            gaUtils.trackEvent("Click", "Mobile Bottom Buttons - Library Card")
          }
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeLibraryCard"
            size="large"
          />
          {upperNavLinks.libraryCard.text}
        </Link>
        <Link
          href={upperNavLinks.emailUpdates.href}
          borderTop="1px solid rgb(54, 54, 54)"
          gridColumn="2 / span 1"
          onClick={() =>
            gaUtils.trackEvent("Click", "Mobile Bottom Buttons - Email Updates")
          }
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeEnvelope"
            size="large"
          />
          {upperNavLinks.emailUpdates.text}
        </Link>
        <Link
          href={upperNavLinks.shop.href}
          borderTop="1px solid rgb(54, 54, 54)"
          gridColumn="1 / span 2"
          onClick={() =>
            gaUtils.trackEvent("Click", "Mobile Bottom Buttons - Shop NYPL")
          }
        >
          <Icon
            align="left"
            color="ui.white"
            name="decorativeShoppingBag"
            size="large"
          />
          {upperNavLinks.shop.text} NYPL
        </Link>
        <Link
          href={upperNavLinks.donate.href}
          gridColumn="1 / span 2"
          onClick={() => gaUtils.trackEvent("Donate", "Mobile Buttons Donate")}
        >
          {upperNavLinks.donate.text.toUpperCase()}
        </Link>
      </SimpleGrid>
    </Box>
  );
});

export default HeaderMobileNav;
