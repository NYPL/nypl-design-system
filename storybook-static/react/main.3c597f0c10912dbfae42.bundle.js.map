{"version":3,"file":"main.3c597f0c10912dbfae42.bundle.js","sources":["webpack:///main.3c597f0c10912dbfae42.bundle.js"],"mappings":"AAAA","sourceRoot":""}