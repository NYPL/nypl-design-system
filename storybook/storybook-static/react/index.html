<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><style>html, body {
    overflow: hidden;
    height: 100%;
    width: 100%;
    margin: 0;
    padding: 0;
  }</style><script>/* globals window */
  /* eslint-disable no-underscore-dangle */
  try {
    if (window.top !== window) {
      window.__REACT_DEVTOOLS_GLOBAL_HOOK__ = window.top.__REACT_DEVTOOLS_GLOBAL_HOOK__;
    }
  } catch (e) {
    // eslint-disable-next-line no-console
    console.warn('unable to connect to top frame for connecting dev tools');
  }</script><style>#root[hidden],
      #docs-root[hidden] {
        display: none !important;
      }</style></head><body><div id="root"></div><div id="docs-root"></div><script>window['DOCS_MODE'] = false;</script><script src="runtime~main.00bd9d033d7a0b972fc9.bundle.js"></script><script src="vendors~main.d2b6aafef1a259d165c0.bundle.js"></script><script src="main.3bf71672a6dd35499004.bundle.js"></script></body></html>