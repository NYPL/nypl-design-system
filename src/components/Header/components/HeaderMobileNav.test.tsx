import { render, screen, within } from "@testing-library/react";

import { axe } from "jest-axe";
import * as React from "react";
import renderer from "react-test-renderer";

import HeaderMobileNav from "./HeaderMobileNav";

describe("HeaderMobileNav Accessibility", () => {
  it("passes axe accessibility test", async () => {
    const { container } = render(<HeaderMobileNav />);
    expect(await axe(container)).toHaveNoViolations();
  });
});

describe("HeaderMobileNav", () => {
  beforeEach(() => {
    render(<HeaderMobileNav />);
  });

  it("renders the NYPL logo", () => {
    const logo = screen.getByRole("img");

    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("title", "NYPL Header Logo");
  });

  it("renders a nav list", () => {
    const navList = screen.getByRole("navigation");
    const links = within(navList).getAllByRole("link");

    expect(links).toHaveLength(7);
    expect(links[0]).toHaveTextContent("Books/Music/Movies");
    expect(links[1]).toHaveTextContent("Research");
    expect(links[2]).toHaveTextContent("Education");
    expect(links[3]).toHaveTextContent("Events");
    expect(links[4]).toHaveTextContent("Connect");
    expect(links[5]).toHaveTextContent("Give");
    expect(links[6]).toHaveTextContent("Get Help");
  });

  it("renders the bottom links", () => {
    const bottomGrid = screen.getByTestId("bottomLinks");
    const links = within(bottomGrid).getAllByRole("link");

    expect(links).toHaveLength(4);
    expect(links[0]).toHaveTextContent("Get a Library Card");
    expect(links[1]).toHaveTextContent("Get Email Updates");
    expect(links[2]).toHaveTextContent("Shop NYPL");
    expect(links[3]).toHaveTextContent("Donate");
  });

  it("renders the UI snapshot correctly", () => {
    const headerMobileNav = renderer.create(<HeaderMobileNav />).toJSON();

    expect(headerMobileNav).toMatchSnapshot();
  });
});
