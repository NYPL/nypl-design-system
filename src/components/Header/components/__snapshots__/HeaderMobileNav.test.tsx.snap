// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMobileNav renders the UI snapshot correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-k008qs"
  >
    <div
      className="css-0"
    >
      <svg
        aria-hidden={false}
        aria-label="NYPL Header Logo"
        className="chakra-icon css-1grhd2q"
        focusable={false}
        role="img"
        title="NYPL Header Logo"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
    </div>
    <div
      className="css-17xejub"
    />
    <nav
      aria-label="Header mobile links"
    >
      <ul
        className="css-1xdhyk6"
        id="header-mobile-nav"
      >
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/books-music-movies"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Books/Music/Movies
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/research"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Research
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/education"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Education
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/events"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Events
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/connect"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Connect
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/give"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Give
          </a>
        </li>
        <li>
          <a
            className="css-1xdhyk6"
            href="https://www.nypl.org/get-help"
            onClick={[Function]}
            rel={null}
            target={null}
          >
            Get Help
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="css-wcncoc"
    data-testid="bottomLinks"
  >
    <a
      className="css-g8syzw"
      href="https://www.nypl.org/library-card/new"
      onClick={[Function]}
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1g9tbrc"
        focusable={false}
        role="img"
        title="decorativeLibraryCard icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Get A Library Card
    </a>
    <a
      className="css-1i5qkun"
      href="https://pages.email.nypl.org/page.aspx?QS=3935619f7de112ef7250fe02b84fb2f9ab74e4ea015814b7"
      onClick={[Function]}
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1g9tbrc"
        focusable={false}
        role="img"
        title="decorativeEnvelope icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Get Email Updates
    </a>
    <a
      className="css-1hbu0qa"
      href="https://shop.nypl.org/?utm_campaign=NYPLHeaderButton&utm_source=nypl.org&utm_medium=referral"
      onClick={[Function]}
      rel={null}
      target={null}
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-1g9tbrc"
        focusable={false}
        role="img"
        title="decorativeShoppingBag icon"
        viewBox="0 0 24 24"
      >
        <g
          stroke="currentColor"
          strokeWidth="1.5"
        >
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokeLinecap="round"
          />
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokeLinecap="round"
          />
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokeMiterlimit="10"
          />
        </g>
      </svg>
      Shop
       NYPL
    </a>
    <a
      className="css-847rpw"
      href="https://secure.nypl.org/site/Donation2?7825.donation=form1&df_id=7825&mfc_pref=T&s_src=FRQ18ZZ_TNN"
      onClick={[Function]}
      rel={null}
      target={null}
    >
      DONATE
    </a>
  </div>
</div>
`;
